.github/workflows/ci.yml

```diff
@@ -81,3 +81,10 @@ jobs:
       - name: Run E2E Tests
         shell: bash
         run: AVALANCHEGO_BUILD_PATH=/tmp/e2e-test/avalanchego DATA_DIR=/tmp/e2e-test/data ./scripts/run_ginkgo.sh
+      - name: Upload Artifact
+        if: always()
+        uses: actions/upload-artifact@v3
```

Comment:
do we always want this step to run even if another step fails?

Reply:
This is intended to run in the case of failure so that we can view the logs of the ANR from an e2e test failure. It would be great to run this iff `Run E2E Tests` runs, but not sure of a cleaner way to accomplish exactly that.